src/app/content/practiceQuestions/components/PracticeQuestionsPopup.tsx

```diff
@@ -22,15 +23,18 @@ const PracticeQuestionsPopup = () => {
   const currentQuestionIndex = useSelector(pqSelectors.currentQuestionIndex);
   const bookTheme = useSelector(bookThemeSelector);
   const intl = useIntl();
+  const match = useSelector(navigation.match);
 
   const closeAndTrack = React.useCallback((method: string) => () => {
     if (currentQuestionIndex !== null) {
       const message = intl.formatMessage({ id: 'i18n:practice-questions:popup:warning-before-close' });
       if (!assertWindow().confirm(message)) { return; }
     }
-    dispatch(closePracticeQuestions());
+    if (match) {
+      dispatch(goBack(match));
+    }
```

Comment:
it is unclear to me why we want this dispatch being called only when `match` is truth?
Also, I think it shouldn't be `goBack` action and instead, it should be `PUSH`.
Let's imagine the case when the user went to page A where he found a link to the page with a modal query in a URL
After clicking this link he will go to the page and modal will be opened. Then the user clicks X icon and he goes back to the page where he started, but what he really wanted was to close the modal.
Instead of going back we should remove `modal=PQ` from the current query and call PUSH with the remaining URL

Reply:
Now when it is changed to `push`, it cannot be called without `match`. So i think that the `if` statement should stay.

Reply:
Can `match` not be `truthy` when `closeAndTrack` is called?

Reply:
Hmm.. I think not, but I will have to make sure. If it can not be `falsy`, then I'll have to use typescript's non-null exclamation mark, right?

Reply:
yes

Reply:
please use `assertNonNull` helper function